subtrack.MAUI/Services/SubscriptionService.cs

```diff
@@ -11,9 +12,11 @@ internal class SubscriptionService : ISubscriptionService
         private readonly SubtrackDbContext _context;
         public SubscriptionService(SubtrackDbContext context) => _context = context;
 
-        public async Task<IEnumerable<Subscription>> GetSubscriptions()
+        public async Task<IEnumerable<Subscription>> GetSubscriptions(GetSubscriptionsFilter? filter = null)
         {
-            return await _context.Subscriptions.ToListAsync();
+            bool shouldFilter = filter != null;
+
+            return await _context.Subscriptions.WhereIf(filter?.GetAutoPaidFilter(), shouldFilter).ToListAsync();
```

Comment:
Related to making the filter contain all filterings, here you would do .WhereIf(x => x.AutoPaid == filter.AutoPaid, filter.AutoPaid != Null)

So later one we can simply chain on more .WhereIf() when we add more filters